scripts/bmm_examples/Example3_Bays2009.R

```diff
@@ -38,96 +38,98 @@ if (nChains >  parallel::detectCores()) {
 
+# model object
+model_bays_2009 <- mixture3p(resp_error = "RespErr",
+                             nt_features = "col_lure",
+                             set_size = "setsize",
+                             regex = TRUE)
+
 # formula
-ff <- bf(RespErr ~ 1,
-         kappa ~ 0+setsize+(0+setsize || subID),
-         thetat ~ 0+setsize+(0+setsize || subID),
-         thetant ~ 0+setsize+(0+setsize || subID))
+formula_bays_2009 <- bmf(kappa ~ 0 + setsize + (0 + setsize || subID),
+                         thetat ~ 0 + setsize + (0 + setsize || subID),
+                         thetant ~ 0 + setsize + (0 + setsize || subID))
 
 # if the model has been already estimated, load the results, otherwise estimate it
-filename <- 'output/fit_bays2009_3p_model.RData'
-if (!file.exists(here(filename))) {
-  
-  fit_bays2009 <- bmm::fit_model(formula = ff, 
-                                 data = data_Bays2009, 
-                                 model_type = '3p',
-                                 non_targets=paste0('Pos_Lure', 1:5),
-                                 setsize = "setsize",
-                                 parallel = TRUE,
-                                 
-                                 # save settings
-                                 sample_prior = TRUE,
-                                 save_pars = save_pars(all = TRUE),
-                                 
-                                 # add brms settings
-                                 warmup = warmup_samples,
-                                 iter = warmup_samples + postwarmup_samples, 
-                                 chains = nChains,
-                                 
-                                 # control commands for the sampler
-                                 control = list(adapt_delta = adapt_delta, 
-                                                max_treedepth = max_treedepth))  
-  
-  save(list=ls(), file=here(filename))  
-} else load(here(filename))
+filename <- here('output','fit_bays2009_3p_model')
+
+fit_bays_2009 <- bmm(model = model_bays_2009,
+                     formula = formula_bays_2009, 
+                     data = data_bays_2009, 
+                     
+                     # save settings
+                     sample_prior = TRUE,
+                     save_pars = save_pars(all = TRUE),
+                     
+                     # add brms settings
+                     warmup = warmup_samples,
+                     iter = warmup_samples + postwarmup_samples, 
+                     chains = nChains,
+                     cores = parallel::detectCores(),
+                     
+                     # control commands for the sampler
+                     control = list(adapt_delta = adapt_delta, 
+                                    max_treedepth = max_treedepth),
+                     
+                     # save results to file
+                     file = filename)
```

Comment:
"Error: Object loaded via 'file' is not of class 'bmmfit'." - see my comment on Example 1